docker/Dockerfile

```diff
@@ -37,6 +37,7 @@ RUN \
       libbz2-dev \
       libenchant-dev \
       libffi-dev \
+      libicu-dev \
```

Comment:
You might need to separate out changes modifying Dockerfile's from code changes, to make sure that gitlab and circleci use this one.  

Similar to what I did here
https://github.com/DataDog/dd-trace-py/pull/10373 - updating docker/Dockerfile
https://github.com/DataDog/dd-trace-py/pull/10405 - changing references to testrunner image

Reply:
At the end, I removed libicu stuff since it was only being used in one case that could be handled by a conversion to `py::str` without performance loss.